src/mpid/ch4/shm/xpmem/xpmem_avl.h

```diff
@@ -0,0 +1,507 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2006 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ *  Portions of this code were written by Intel Corporation.
+ *  Copyright (C) 2011-2017 Intel Corporation.  Intel provides this material
+ *  to Argonne National Laboratory subject to Software Grant and Corporate
+ *  Contributor License Agreement dated February 8, 2012.
+ */
+#ifndef XPMEM_AVL_H_INCLUDED
+#define XPMEM_AVL_H_INCLUDED
+
+#include "xpmem_pre.h"
+
+enum {
+    AVL_TREE_RIGHT,
+    AVL_TREE_LEFT
+};
+
+typedef int (*MPIDI_XPMEM_avl_tree_seg_delete_cb_t) (MPIDI_XPMEM_seg_t *);
+typedef int (*MPIDI_XPMEM_avl_tree_seg_create_cb_t) (MPIDI_XPMEM_seg_t **, uint64_t, uint64_t,
+                                                     MPIDI_XPMEM_segmap_t *);
+
+/* AVL tree is balanced tree, so 64 stack size should be enough to go through all the nodes in the tree */
+#define MPIDI_XPMEM_AVL_TREE_STACK_SIZE 64
+#define MPIDI_XPMEM_AVL_TREE_ALLOCATE_STACK(stack, type, size) \
+        const size_t stack##_size = size; \
```

Comment:
Is `MPIDI_XPMEM_AVL_TREE_DECLARE_STACK` more suitable ? Otherwise we might think it allocates memory buffer.